blobmodel/stochasticality.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from .blobs import Blob
+from nptyping import NDArray, Float
+from typing import Any
+
+
+class BlobFactory(ABC):
+    """
+    Abstract class used by 2d propagating blob model to specify blob parameters. 
+    """
+
+    @abstractmethod
+    def sample_blobs(
+        self, Ly: float, T: float, num_blobs: int, blob_shape: str, t_drain: float
+    ) -> list[Blob]:
+        raise NotImplementedError
+
+
+class DefaultBlobFactory(BlobFactory):
+    """
+    Defualt blob parameters 
+    """
+
+    def __init__(
+        self,
+        A_dist: str = "exp",
+        W_dist: str = "exp",
+        vx_dist: str = "deg",
+        vy_dist: str = "normal",
+        A_parameter: float = 1.0,
+        W_parameter: float = 1.0,
+        vx_parameter: float = 1.0,
+        vy_parameter: float = 1.0,
+    ) -> None:
+        self.A_dist = A_dist
+        self.W_dist = W_dist
+        self.vx_dist = vx_dist
+        self.vy_dist = vy_dist
+        self.A_parameter = A_parameter
+        self.W_parameter = W_parameter
+        self.vx_parameter = vx_parameter
+        self.vy_parameter = vy_parameter
+
+    def __choose_distribution(
+        self, dist_type: str, free_parameter: float, num_blobs: int,
+    ) -> NDArray[Any, Float[64]]:
+        """
+        The following distributions are implemented:
+            exp: exponential distribution with mean 1
+            gamma: gamma distribution with `free_parameter` as shape parameter and mean 1
+            normal: normal distribution with zero mean and `free_parameter` as scale parameter
+            uniform: uniorm distribution with mean 1 and `free_parameter` as width
+            ray: rayleight distribution with mean 1
+            deg: array on ones
+            zeros: array of zeros
+        """
```

Comment:
I like the solution with this free parameter. However, this docstring should be in the constructor: The user does not need to know about the private __choose_distribution function, which he will never call. Instead, he will interact with the constructor.

Reply:
I moved the docstring to the constructor